penguin/llm/adapters/openrouter.py

```
@@ -2076,19 +2078,43 @@ def get_last_usage(self) -> Dict[str, Any]:
+                if isinstance(tool_call, dict) and tool_call.get("name")
+            ]
+            if not pending and isinstance(
+                getattr(self, "_last_tool_call", None), dict
+            ):
+                pending = [dict(self._last_tool_call)]
+            self._pending_tool_calls = []
             self._last_tool_call = None
             self._tool_call_acc = {"name": None, "arguments": ""}
-            return dict(data) if isinstance(data, dict) else None
+            self._tool_call_accs = {}
+            return pending
```

Ruff (0.15.11)

[warning] Consider moving this statement to an `else` block

(TRY300)